pkgs/io_file/lib/src/file_system.dart

```diff
@@ -24,3 +29,11 @@ base class FileSystem {
     throw UnsupportedError('rename');
   }
 }
+
+FileSystem get fileSystem {
```

Comment:
Perhaps:

```
FileSystem get fileSystem => Platform.isWindows ? WindowsFileSystem() : PosixFileSystem();
```

I assume creating these is cheap?

Reply:
Yep. These classes have no state.